### PR TITLE
Add more status symbols to tab list

### PR DESCRIPTION
**Pull request by fluem**
Issues #1338 and #2811 
Introduces tab status symbols ```P```,```A```,```M```,```D``` for pinned, audible, muted, and discarded tabs. If ```completions.Tab.statusstylepretty``` is set to ```true```, capital letters are replaced with Unicode characters. The plain uppercase letters are kept in a hidden column, so that they can still be used for filtering.

Filtering tabs by using ```P```,```A```,```M```,```D``` works whether ```statusstylepretty``` is set or not, although there are still some cases where I don't understand the filtering; will try to work these out.

### CODE REVIEW

File: src/completions/Tab.ts [hunk 1]
```diff
@@ -24,17 +24,33 @@ class BufferCompletionOption
...
+            if (tab.audible) pre += "\uD83D\uDD0A"
+            if (tab.mutedInfo.muted) pre += "\uD83D\uDD07"
+            if (tab.discarded) pre += "\u2296"
+        } else {
+            pre = preplain
+        }
 
         // Push prefix before padding so we don't match on whitespace
         this.fuseKeys.push(pre)
+        this.fuseKeys.push(preplain)
```
Comment (bovine3dom) [inline src/completions/Tab.ts:53]: Hmm, if we do this it won't do the case sensitive matching.

I was imagining that we would do something more like the matching we do for numbers.

Comment (fluem) [reply src/completions/Tab.ts:53]: Yes, agreed. I don't fully understand how the completion code works yet. Testing out some combinations of tabs does not always give me the selection I expected.

**Comment by bovine3dom** [comment]
I really thought I had merged this and was wondering why I wasn't getting the symbols in the tab list, I assumed something had broken it and was about to file an issue...

Bumping it in the hope that I will remember to review and merge it soon

**Comment by bovine3dom** [comment]
LGTM, just waiting for the tests to pass. Sorry it took such an absurd amount of time to review!

**Pull request merged**